base/kra/src/main/java/com/netscape/cmscore/dbs/KeyRepository.java

```diff
@@ -547,69 +590,35 @@ public KeyRecordList findKeyRecordsInList(String filter,
     public BigInteger getLastSerialNumberInRange(BigInteger serial_low_bound, BigInteger serial_upper_bound) throws
             EBaseException {
 
-        logger.debug("KeyRepository:  in getLastSerialNumberInRange: low "
-                + serial_low_bound + " high " + serial_upper_bound);
+        logger.debug("KeyRepository:  in getLastSerialNumberInRange: low {} high {}",
+                serial_low_bound, serial_upper_bound);
 
         if (serial_low_bound == null
                 || serial_upper_bound == null || serial_low_bound.compareTo(serial_upper_bound) >= 0) {
             return null;
         }
 
-        String ldapfilter = "(" + "serialno" + "=*" + ")";
+        String ldapfilter = "(&(" + "serialno" + "=*" + ")(" + KeyRecord.ATTR_ID + "<="+serial_upper_bound+"))";
```

Comment:
Should we use `serialno` or `keySerialNumber` in this filter? I'm not too familiar with this area, but sometimes the filter needs to be defined using the record attributes (e.g `keySerialNumber`) and sometimes it needs to be defined using the actual LDAP attributes (.e.g `serialno`). The mapping is defined here:
https://github.com/dogtagpki/pki/blob/master/base/kra/src/main/java/com/netscape/cmscore/dbs/KeyRepository.java#L101-L103

Reply:
In this case it is OK to use `serialno` because it is the name of the attribute stored in the DS entry and it has to collect everything. The  `keySerialNumber` is derived from `serialno` and contain the real serial number to use (the value stored in the DS has a special format which allow to order by `serialno` even it is a string).
Nevertheless, I totally agree that we can remove this condition since it is always valid. I have done some test without and it is OK.